src/uct/tcp/tcp_iface.c

```diff
@@ -255,10 +257,28 @@ static void uct_tcp_iface_connect_handler(int listen_fd, void *arg)
 
         fd = accept(iface->listen_fd, (struct sockaddr*)&peer_addr, &addrlen);
         if (fd < 0) {
+            if (errno == ECONNABORTED) {
+                ucs_warn("peer destroyed a connection right after it was established");
+                return;
+            }
+
             if ((errno != EAGAIN) && (errno != EWOULDBLOCK) && (errno != EINTR)) {
                 ucs_error("accept() failed: %m");
                 uct_tcp_iface_listen_close(iface);
             }
+
+            return;
+        }
+
+        ucs_assert(addrlen == sizeof(peer_addr));
+
+        if (!ucs_socket_is_connected(fd)) {
```

Comment:
maybe better move this check inside uct_tcp_cm_handle_incoming_conn()?

Reply:
done